src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -582,12 +582,8 @@ static const int fast_log2 [] = { -1, -1, 1, -1, 2, -1, -1, -1, 3 };
 static MonoInst*
 emit_sum_vector (MonoCompile *cfg, MonoType *vector_type, MonoTypeEnum element_type, MonoInst *arg)
 {
-	MonoClass *arg_class = mono_class_from_mono_type_internal (vector_type);
-	int size = mono_class_value_size (arg_class, NULL);
-	if (size != 16) 		// Works only with Vector128
```

Comment:
This removes the check for 256/512 bytes as well.

Reply:
Yes, but at the moment we emit SIMD intrinsics only for Vector128 on amd64. I could keep the checks for the future when we support Vector256/512?

Reply:
I think we should keep them.

Reply:
Currently Vector256/512 won't hit this code path. I prefer removing the check. We will need to re-evaluate the generated code anyway when enabling support Vector256/512.

Reply:
Additionally, `emit_sri_vector` is also guarded by
https://github.com/dotnet/runtime/blob/main/src/mono/mono/mini/simd-intrinsics.c#L1125-L1126

Thus, I don't have any concern removing these checks.